### PR TITLE
Create release-drafter.yml

### PR DESCRIPTION
**Pull request by cliu123**
*Issue #, if available:*

*Description of changes:*
In order to avoid going through all the merged PRs and generate release notes manually at the time of release, release-drafter is introduced to keep track of the changes and generate release notes automatically as and when each PR gets merged into master branch

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by vrozov** [comment]
Please provide an example of the test run.

**Comment by sujithvm** [comment]
- Can we add template also to categorize into various categories Enhancements, Bug fixes, maintenance ? We already have labels for them 

- We create draft release as part of CD workflow https://github.com/opendistro-for-elasticsearch/security/blob/master/.github/workflows/cd.yml#L67 Is it possible to fetch release drafter output url and then upload artifacts to that draft ?

File: release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Release-drafter
+
+on:
+#   push:
+#     branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (sujithvm) [inline release-drafter.yml:7]: Should this be push on master branch which would run after a pull request is squashed and merged?

File: release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+   update_release_draft:
+    runs-on: ubuntu-latest
+    steps:
+      # Drafts the next Release notes as Pull Requests are merged into "master"
+      - uses: release-drafter/release-drafter@v5
+        # with:
+          # (Optional) specify config name to use, relative to .github/. Default: release-drafter.yml
+          # config-name: my-config.yml
```
Comment (sujithvm) [inline release-drafter.yml:17]: Can we remove these comments ?

File: release-drafter.yml (hunk 1 above)
Comment (cliu123) [reply release-drafter.yml:7]: Resolved.

File: release-drafter.yml (hunk 2 above)
Comment (cliu123) [reply release-drafter.yml]: Resolved.

File: release-drafter.yml (hunk 1 above)
Comment (sujithvm) [reply release-drafter.yml:7]: Are both the trigger needed ?

Comment (cliu123) [reply release-drafter.yml:7]: I believe so. When a PR gets checked in on master branch, the change should also be reflected in release notes.

Comment (sujithvm) [reply release-drafter.yml:7]: Can you validate this behavior? AFAIK pull_request will trigger when pull request is created against the master branch. We want the release notes to be updated only after the pull request is merged to master (which results in a push on master branch). 

Check release drafter: https://github.com/release-drafter/release-drafter#usage

**Pull request closed without merging**